app/controllers/supplementary/DocumentsProducedController.scala

```diff
@@ -141,7 +155,7 @@ class DocumentsProducedController @Inject()(
     if (cachedData.containsItem(docToRemove)) {
       val updatedCache = cachedData.copy(documents = cachedData.documents.filterNot(_ == docToRemove))
 
-      customsCacheService.cache[DocumentsProducedData](supplementaryCacheId, formId, updatedCache).map { _ =>
+      customsCacheService.cache[DocumentsProducedData](goodsItemCacheId, formId, updatedCache).map { _ =>
```

Comment:
In every place you use `customsCacheService` but in add goods item you use this `items` one. Probably everything in this file is okay, but I want to understand how data from this screen is handled? I mean for me when we discussed about separate cache I thought we will be storing all item data in separate cache and when use finished adding item we will add this to the main cache clear item cache and give a possibility to the user to go through the journey once again.

Reply:
The goodsItemCacheId is the temp cache for adding item elements. if you looks at itemsCache you will get an understanding